src/Microdown-BookTester/MicReferenceChecker.class.st

```diff
@@ -72,8 +72,21 @@ MicReferenceChecker >> addDuplicatedFirstAnchor: anAnchor [
 { #category : 'main API' }
 MicReferenceChecker >> checkDirectory: aDir [
 	"Take the directory, parse all its children with microdown file parser and let the visitor visit each time then return visitor is ok which should be true if every thing is okay, the visitor turned out to treat the many documents that it visits as one, so if anchor is duplicated in another file it will detect that . "
-	
-	self checkList: aDir allFiles
```

Comment:
I asked myself why we do not call checkList:
Another question I asked my self was why the collector is not shared between the checkDirectory: and checkProject:

Reply:
Here I reused the checkProject: code for having the same semantic in both of checkProject and checkDirectory. chekList: is call at the end of the method.

For the collector who aren't shared between the checkDirectory: and checkProject:, I didn't think about it, maybe I can create a collector variable for the class